docs/source/reducing_memory_usage.md

```diff
@@ -268,6 +268,190 @@ training_args = RLOOConfig(..., ds3_gather_for_generation=False)
 
 This adjustment prevents model weights from being gathered, avoiding OOM errors, but it may result in slower generation speeds.
 
+## Context Parallelism
+
+Context Parallelism (CP) is a parallelization technique that enables training with longer sequences by splitting the sequence dimension across multiple GPUs. Each GPU processes a portion of the sequence, allowing you to train with sequences longer than what would fit on a single GPU's memory.
+
+For more details on Context Parallelism, see the [Ultrascale Playbook - Context Parallelism](https://huggingface.co/spaces/nanotron/ultrascale-playbook?section=context_parallelism).
+
+Context parallelism is particularly useful when:
+- You want to train with very long sequences (>2048 tokens)
+- Single GPU memory is insufficient for your desired sequence length
+- You need to maintain sequence coherence across the full context
+
+### Requirements and Limitations
+
+Context Parallelism has specific requirements:
+
+1. **FSDP2 (PyTorch FSDP v2)** is required as the distributed training backend
+2. **SDPA attention** - Flash Attention is currently not supported with Context Parallelism
+3. **Sequence length divisibility** - sequences must be divisible by `cp_size * 2`
+
+### Configuration
+
+To enable Context Parallelism, you need to configure both Accelerate and your training arguments:
+
+#### Accelerate Configuration
+
+Create an accelerate config file (e.g. `context_parallel_config.yaml` for 2 GPUs):
+
+```yaml
+compute_environment: LOCAL_MACHINE
+debug: false
+distributed_type: FSDP
+downcast_bf16: 'no'
+enable_cpu_affinity: false
+fsdp_config:
+  fsdp_activation_checkpointing: false
+  fsdp_auto_wrap_policy: TRANSFORMER_BASED_WRAP
+  fsdp_cpu_ram_efficient_loading: true
+  fsdp_offload_params: false
+  fsdp_reshard_after_forward: true
+  fsdp_state_dict_type: FULL_STATE_DICT
+  fsdp_version: 2
+machine_rank: 0
+main_training_function: main
+mixed_precision: bf16
+num_machines: 1
+num_processes: 2  # Number of GPUs
+rdzv_backend: static
+same_network: true
+tpu_env: []
+tpu_use_cluster: false
+tpu_use_sudo: false
+use_cpu: false
+parallelism_config:
+  parallelism_config_dp_replicate_size: 1
+  parallelism_config_dp_shard_size: 1
+  parallelism_config_tp_size: 1
+  parallelism_config_cp_size: 2  # Context parallel size
+```
+
+#### Training Configuration
+
+<hfoptions id="context-parallel">
+<hfoption id="SFT">
+
+You can configure context parallelism training either programmatically or via command line:
+
+**Option 1: Using SFTConfig**
+
+**With Wrapped Strategy:**
```

Comment:
I recommend not mentionning `"wrapped"` in the documentation. I feel like this is a very advanced use, and it somewhat distracts from the message of this section.